sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubMessageSerializer.java

```diff
@@ -107,14 +118,80 @@ public int getSize(Message amqpMessage) {
 
         final EventData eventData = (EventData) object;
         final Message message = Proton.message();
+        final byte[] body = eventData.getBody();
+
+        //TODO (conniey): support AMQP sequence and AMQP value.
+        message.setBody(new Data(new Binary(body)));
 
         if (eventData.getProperties() != null && !eventData.getProperties().isEmpty()) {
             message.setApplicationProperties(new ApplicationProperties(eventData.getProperties()));
         }
 
-        setSystemProperties(eventData, message);
+        if (message.getProperties() == null) {
+            message.setProperties(new Properties());
+        }
+
+        message.setMessageId(eventData.getMessageId());
+        message.setContentType(eventData.getContentType());
+        message.setCorrelationId(eventData.getCorrelationId());
+
+        final AmqpMessageProperties brokeredProperties = eventData.getRawAmqpMessage().getProperties();
+
+        message.setContentEncoding(brokeredProperties.getContentEncoding());
```

Comment:
We seem to be missing some properties defined by the AMQP spec  _([ref](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-properties))_:

- Group Id
- ReplyTo
- ReplyToGroupId
- Subject
- To

Reply:
@jsquire 1. Do the above five attributes have `Getter` and `Setter` operations ?  2. Does not the `Group Id` performing verification such as `checkIdLength` ?

Reply:
These attributes already exist in the correct form on the [AmqpMessageProperties](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpMessageProperties.java) exposed by the `AmqpAnnotatedMessage`.    We should not be adding any members on `EventData` to access them directly; they're available via `getRawAmqpMessage()` only.

My comment was meant to call out that we're not including them in serialization here.  We should be reading the values from the `brokeredProperties` into the corresponding `message.getProperties()` attributes without doing any kind of validation on them.

Reply:
According to your instructions, fix in new version.